osquery/config/plugins/filesystem.cpp

```diff
@@ -33,18 +38,68 @@ REGISTER(FilesystemConfigPlugin, "config", "filesystem");
 std::pair<osquery::Status, std::string> FilesystemConfigPlugin::genConfig() {
   std::string config;
   if (!fs::exists(FLAGS_config_path)) {
-    return std::make_pair(Status(1, "config file does not exist"), config);
+    return std::make_pair(Status(1, "config file does not exist"), "");
   }
+  std::vector<std::string> conf_files;
+  Status stat = resolveFilePattern(FLAGS_config_path + ".d/%.conf", conf_files);
+  if (!stat.ok()) {
+    VLOG(1) << "Error is resolving extra configuration files: "
+            << stat.getMessage();
+  }
+  VLOG(1) << "Finished resolving, merging " << conf_files.size()
+          << " additional JSONs";
+
+  std::sort(conf_files.begin(), conf_files.end());
+  conf_files.push_back(FLAGS_config_path);
+  pt::ptree merged, scheduled_queries, options, additional_monitoring;
+
+  for (const auto& conf_file : conf_files) {
+    VLOG(1) << "Filesystem ConfigPlugin reading: " << conf_file;
 
-  VLOG(1) << "Filesystem ConfigPlugin reading: " << FLAGS_config_path;
-  std::ifstream config_stream(FLAGS_config_path);
+    std::ifstream config_stream(conf_file);
+    config_stream.seekg(0, std::ios::end);
+    config.reserve(config_stream.tellg());
+    config_stream.seekg(0, std::ios::beg);
 
-  config_stream.seekg(0, std::ios::end);
-  config.reserve(config_stream.tellg());
-  config_stream.seekg(0, std::ios::beg);
+    config.assign((std::istreambuf_iterator<char>(config_stream)),
+                  std::istreambuf_iterator<char>());
 
-  config.assign((std::istreambuf_iterator<char>(config_stream)),
-                std::istreambuf_iterator<char>());
-  return std::make_pair(Status(0, "OK"), config);
+    std::stringstream json;
+    json << config;
+
+    pt::ptree tree;
+    pt::read_json(json, tree);
+    for (const pt::ptree::value_type& v : tree.get_child("scheduledQueries")) {
```

Comment:
Can we make this merge-related code available to any config plugin? Seems a shame to keep it in the filesystem-specific plugin.